app/data_model/answer_store.py

```diff
@@ -247,6 +254,26 @@ def get_hash(self):
         """
         return hash(json.dumps(self.answers, sort_keys=True))
 
+    def upgrade(self):
+        if self.version < self._LATEST_VERSION:
+            if self.version < 1:
+                # Update Date formats
+                for answer in self.answers:
+                    block_json = SchemaHelper.get_block(g.schema_json, answer['block_id'])
+                    answer_schema_list = SchemaHelper.get_answers_by_id_for_block(block_json)
+                    answer_schema = answer_schema_list[answer['answer_id']]
```

Comment:
I think this logic should live in SchemaHelper, so we could call something like `SchemaHelper.get_answer_schema_for_answer_id(answer['answer_id'])`, or maybe even going as far as a `get_answer_type` method in SchemaHelper.